src/pages/keystone-test.tsx

```diff
@@ -0,0 +1,419 @@
+  addCollectionsInput,
+} from 'operations/mutations/addCollections'
+import { useAddBookmarkMutation } from 'operations/mutations/addBookmark'
+import { useAddCollectionMutation } from 'operations/mutations/addCollection'
+import { useAnalytics } from 'stores/analyticsContext'
+import { client } from 'apolloClient'
+
+type SortBy = 'SORT_TYPE' | 'SORT_ALPHA'
+
+type SelectedCollections = string[]
+
+/** Type guards */
+function isCollection(widget: MySpaceWidget): widget is CollectionType {
+  return widget.type === WIDGET_TYPES.COLLECTION
+}
+
+const KeystoneTest = ({
+  collections,
+  bookmarks,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) => {
+  const router = useRouter()
+  const { user } = useUser()
+  const { loading, error, data } = useMySpaceQuery()
+  const { trackEvent } = useAnalytics()
+
+  const [sortBy, setSort] = useState<SortBy>('SORT_TYPE')
+  const [selectMode, setSelectMode] = useState<boolean>(false)
+  const [selectedCollections, setSelectedCollections] =
+    useState<SelectedCollections>([])
+  const [flash, setFlash] = useState<React.ReactNode>(null)
+
+  const [handleAddCollections] = useAddCollectionsMutation()
+  const [handleAddCollection] = useAddCollectionMutation()
+  const [handleAddBookmark] = useAddBookmarkMutation()
+
+  useEffect(() => {
+    if (router.query.selectMode == 'true') {
+      setSelectMode(true)
+    }
+  }, [router.query])
+
+  if (error) return <p>Error</p>
+
+  const userCollections: CollectionType[] =
+    (data &&
+      data.mySpace.filter((w): w is CollectionType => isCollection(w))) ||
+    []
+  const collectionsLength = userCollections.length || 0
+
+  const remainingSections =
+    MAXIMUM_COLLECTIONS - (collectionsLength + selectedCollections.length)
+
+  const canAddSections = collectionsLength < MAXIMUM_COLLECTIONS
+
+  const handleSortClick = (sortType: SortBy) => {
+    const sortTypeAction =
+      sortType === 'SORT_TYPE' ? 'Sort by type' : 'Sort alphabetically'
+    trackEvent('S&A sort', sortTypeAction)
+    setSort(sortType)
+  }
+
+  const handleToggleSelectMode = () => {
+    setSelectMode((currentMode) => !currentMode)
+    setSelectedCollections([])
+  }
+
+  const widgetClasses = classnames(styles.widgetContainer, {
+    [styles.selectMode]: selectMode,
+  })
+
+  const handleSelectCollection = (id: string): void => {
+    if (isSelected(id)) {
+      setSelectedCollections((state) => {
+        const itemIndex = state.indexOf(id)
+        const newState = [...state]
+        newState.splice(itemIndex, 1)
+        return newState
+      })
+    } else {
+      setSelectedCollections((state) => [...state, id])
+    }
+  }
+
+  const isSelected = (id: string): boolean =>
+    selectedCollections.indexOf(id) > -1
+
+  const handleAddSelected = () => {
+    // Find selected collections in the array of all CMS collections
+    const collectionObjs = selectedCollections.map((id) =>
+      collections.find((i) => i.id === id)
+    ) as CollectionRecordInput[]
```

Comment:
I think if you make the type changes suggested, you won't need to cast this using `as CollectionRecordInput` - because at this point, this data is not actually of that type, and is actually `CollectionRecords`